superset/connectors/sqla/models.py

```diff
@@ -677,13 +677,25 @@ def external_metadata(self) -> List[Dict[str, str]]:
             for col in cols:
                 try:
                     if isinstance(col["type"], TypeEngine):
-                        col["type"] = db_engine_spec.column_datatype_to_string(
+                        db_type = db_engine_spec.column_datatype_to_string(
                             col["type"], db_dialect
                         )
+                        type_spec = db_engine_spec.get_column_spec(db_type)
+                        col.update(
+                            {
+                                "type": db_type,
+                                "generic_type": type_spec.generic_type
```

Comment:
I believe this is called `type_generic` on the data model: https://github.com/apache/superset/blob/b6f00e69e156098e9e1569d11f0f7831bac1024f/superset/connectors/base/models.py#L568-L578

Reply:
good catch, rename it. btw, the method `external_metadata(self)` is for getting metadata from the table in database(In other words, not in superset saved metadata).